### PR TITLE
fix(pre-commit): align pre-commit and CI

### PR DESCRIPTION
**Pull request by vringar**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -30,3 +26,10 @@ repos:
         files: 'Explorer/.*\.(js|jsx|ts|tsx)$'
         entry: bash -c 'cd Explorer && npm run precommit-lint "$@"'
         types: [file]
+      - id: golangci-lint
+        name: golangci-lint
+        language: system
+        types: [file, go]
+        # This checks all files instead of only the ones that were changed.
```
Comment (lifesbest23) [inline .pre-commit-config.yaml:33]: This should mean the output of the hook should not differ between `pre-commit run golangci-lint` and `pre-commit run golangci-lint -a` should be the same, which is not the case  for me.

**Comment by lifesbest23** [comment]
When running pre-commit with `-a` I get the following Error message in between:
```
...
Error: parallel golangci-lint is running
Usage:
  golangci-lint run [flags]
...
      --version                   Print version

failed executing command with error parallel golangci-lint is running
...
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -30,3 +26,10 @@ repos:
...
+        # If this check is taking too long we can refine this
+        entry: bash -c 'echo "$@"; RESULT=0;cd Proxy && for i in "$@"; do echo "$i" | cut -d'/' -f2- | xargs golangci-lint run --fix; ((RESULT|=$?)); done && exit $RESULT'
```
Comment (smnws) [inline .pre-commit-config.yaml:35]: This approach does not work with golangci-lint modules like typecheck. Is there a reason to not just run golangci-lint on the whole go package?

**Review by smnws — CHANGES_REQUESTED** [review]
`pre-commit run --all` does report new errors

File: .pre-commit-config.yaml (hunk 2 above)
Comment (vringar) [reply .pre-commit-config.yaml]: Yeah, that's a better idea

**Review by smnws — APPROVED** [review]

**Pull request merged**